src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIProxy.cs

```diff
@@ -782,20 +818,8 @@ private bool InferConnectionDetails()
                         : tokensByCommaAndSlash[1].Trim();
 
                 // Bad Data Source like "server, "
-                if (string.IsNullOrEmpty(parameter))
-                {
-                    ReportSNIError(SNIProviders.INVALID_PROV);
-                    return false;
-                }
-
-                // For Tcp and Only Tcp are parameters allowed.
-                if (ConnectionProtocol == DataSource.Protocol.None)
-                {
-                    ConnectionProtocol = DataSource.Protocol.TCP;
-                }
-                else if (ConnectionProtocol != DataSource.Protocol.TCP)
+                if (string.IsNullOrWhiteSpace(parameter))
```

Comment:
Why was this changed? The statement above trims the parameter anyhow. No whitespaces are expected.